affordance_primitives/test/test_task_monitor.cpp

```diff
@@ -0,0 +1,194 @@
+//          THAT MIGHT OTHERWISE ARISE FROM COURSE OF DEALING OR USAGE OF TRADE.
+//          NO WARRANTY IS EITHER EXPRESS OR IMPLIED WITH RESPECT TO THE USE OF
+//          THE SOFTWARE OR DOCUMENTATION. Under no circumstances shall the
+//          University be liable for incidental, special, indirect, direct or
+//          consequential damages or loss of profits, interruption of business,
+//          or related expenses which may arise from use of software or
+//          documentation, including but not limited to those resulting from
+//          defects in software and/or documentation, or loss or inaccuracy of
+//          data of any kind.
+//
+///////////////////////////////////////////////////////////////////////////////
+
+#include <gtest/gtest.h>
+#include <ros/ros.h>
+
+#include <affordance_primitives/task_monitor/task_monitor.hpp>
+#include <affordance_primitives/msg_types.hpp>
+
+#include <thread>
+
+// constexpr double EPSILON = 1e-4;
+
+using affordance_primitives::ExecutionResult;
+
+TEST(TaskMonitor, start_stop_monitor)
+{
+  ros::NodeHandle nh;
+
+  // Default parameters
+  affordance_primitives::AffordanceParameter params;
+
+  // Create and start monitor with no timeout
+  affordance_primitives::TaskMonitor monitor(nh, "ft_topic_name");
+  ASSERT_NO_THROW(monitor.startMonitor(params));
+
+  // Wait a sec, then stop monitor
+  ros::Duration(0.25).sleep();
+  ASSERT_NO_THROW(monitor.stopMonitor());
+
+  // Result should be STOP_REQUESTED
+  std::optional<ExecutionResult> result = monitor.getResult();
+  ASSERT_TRUE(result.has_value());
+  EXPECT_EQ(result.value(), affordance_primitives::ExecutionResult::STOP_REQUESTED);
+
+  // Now try interupting a timed run
+  ASSERT_NO_THROW(monitor.startMonitor(params, 0.25));
+  ASSERT_NO_THROW(monitor.stopMonitor());
+  result = monitor.getResult();
+  ASSERT_TRUE(result.has_value());
+  EXPECT_EQ(result.value(), affordance_primitives::ExecutionResult::STOP_REQUESTED);
+}
+
+TEST(TaskMonitor, monitor_timing)
+{
+  ros::NodeHandle nh;
+  const double test_duration = 0.25;
+
+  // Default parameters
+  affordance_primitives::AffordanceParameter params;
+
+  // Create and start monitor with no timeout
+  affordance_primitives::TaskMonitor monitor(nh, "ft_topic_name");
+  ASSERT_NO_THROW(monitor.startMonitor(params, test_duration));
+
+  // Immediately getting the result shouldn't work
+  EXPECT_FALSE(monitor.getResult().has_value());
+
+  // Wait some time and grab result
+  ros::Duration(1.05 * test_duration).sleep();
+  std::optional<ExecutionResult> result = monitor.getResult();
+  ASSERT_TRUE(result.has_value());
+  EXPECT_EQ(result.value(), affordance_primitives::ExecutionResult::TIME_OUT);
+
+  // Now test blocking call to waitForResult()
+  ASSERT_NO_THROW(monitor.startMonitor(params, test_duration));
+  ExecutionResult result2;
+
+  ros::Time start = ros::Time::now();
+  result2 = monitor.waitForResult();
+  EXPECT_LE((ros::Time::now() - start).toSec(), 1.05 * test_duration);
```

Comment:
Do we want to make `1.05` a variable, in case we would want to change it at another time

Reply:
I actually think it is better to leave it: if this test fails, gtest will exactly display this line and the dev will be able to see it in the output. And regardless if it is a variable, it means a change in the C++ file